app/src/main/java/com/github/kiolk/cowsandbulls/logic/CustomTimer.java

```diff
@@ -0,0 +1,86 @@
+package com.github.kiolk.cowsandbulls.logic;
+
+import android.util.Log;
+
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class CustomTimer {
+    private long time;
+    private Timer timer;
+    private TimerTask timerTask;
+    private boolean isRunning = false;
+    private TimerChange timerChange;
+
+    public CustomTimer(TimerChange timerChangeListener) {
+        this.timerChange = timerChangeListener;
+        this.time = -1;
+        this.timer = new Timer();
+        this.timerTask = new TimerTask() {
+            private volatile long currentTime = -1;
+
+            @Override
+            public void run() {
+                currentTime++;
+                time = currentTime;
+                timerChange.updateView(getTime());
+            }
+        };
+    }
+
+    public void start() {
+        if (!isRunning) {
+            isRunning = true;
+            this.timer.scheduleAtFixedRate(timerTask, 0, 1000);
+
+        }
+
+
+    }
+
+    public void reset() {
+        this.isRunning = false;
+        this.timerTask.cancel();
+        this.time=-1;
+        this.timerChange.updateView(String.format("%02d:%02d", 0 / 60, 0 % 60));
+        this.timerTask = new TimerTask() {
+            private volatile long currentTime = -1;
+
+            @Override
+            public void run() {
+                currentTime++;
+                time = currentTime;
+                timerChange.updateView(getTime());
+
+            }
+        };
+
+
+    }
+
```

Comment:
Need add two method for handle rotation change. Method must look like 

 public Bundle saveState(Bundle bundle){
        //TODO save need values in bundle
        return bundle;
    }
    
    public CustomTimer restoreTimer(Bundle bundle, TimerChange listener){
        //TODO add logic for restore state of timer
        return new CustomTimer(listener);
    }

Reply:
I don't really understand what I need to do in the second method. Should I read the state of needed items (like I did), or create a new instance of CustomTimer class and return that?

Reply:
Yes, it is correct. First need create new instance of timer. Than configure it by previews values. Start timer if need. Last, return this instance. On activity, you can call this methods in onSaveInstantState and onRestoreInstantState to prevent dismiss timer during rotation.